File: Problem Patterns/Array/longest-common-prefix.java
## Problem Link: https://leetcode.com/problems/longest-common-prefix

class Solution {
    public String longestCommonPrefix(String[] strs) {

        String lcp = strs[0];

        for(int i = 1; i < strs.length; i++)
        {
            String temp = "";
            
            for(int j = 0; j < strs[i].length() && j < lcp.length(); j++)
            {
                if(strs[i].charAt(j) != lcp.charAt(j))
                    break;
                    
                temp += lcp.charAt(j);
            }

            lcp = temp;
        }

        return lcp;

        
    }
}

File: Problem Patterns/Array/longest-common-prefix.py
# Problem Link: https://leetcode.com/problems/longest-common-prefix/

class Solution:
    def longestCommonPrefix(self, strs: List[str]) -> str:
        
        lcp = strs[0]

        for i in range(1, len(strs)):
            # print(lcp, strs[i])
            temp = ''
            for j in range(len(strs[i])):
                if j == len(lcp) or strs[i][j] != lcp[j]:
                    break
                else:
                    temp += lcp[j]
            lcp = temp

        return lcp
